yarn-project/ethereum/src/l1_tx_utils.ts

```diff
@@ -633,4 +655,44 @@ export class L1TxUtils {
     const gasConfig = { ...this.config, ..._gasConfig };
     return gasLimit + (gasLimit * BigInt((gasConfig?.gasLimitBufferPercentage || 0) * 1_00)) / 100_00n;
   }
+
+  /**
+   * Attempts to cancel a transaction by sending a 0-value tx to self with same nonce but higher gas prices
+   * @param nonce - The nonce of the transaction to cancel
+   * @param previousGasPrice - The gas price of the previous transaction
+   * @param attempts - The number of attempts to cancel the transaction
+   * @returns The hash of the cancellation transaction
+   */
+  private async attemptTxCancellation(nonce: number, previousGasPrice?: GasPrice, attempts = 1) {
+    const account = this.walletClient.account;
+
+    // Get gas price with higher priority fee for cancellation
+    const cancelGasPrice = await this.getGasPrice(
+      {
+        ...this.config,
+        // Use much higher bump for cancellation to ensure it replaces the original tx
+        priorityFeeRetryBumpPercentage: 300, // 200% bump should be enough to replace any tx
+      },
+      false,
+      Math.max(attempts, 1),
+      previousGasPrice,
+    );
+
+    this.logger?.debug(`Attempting to cancel transaction with nonce ${nonce}`, {
+      maxFeePerGas: formatGwei(cancelGasPrice.maxFeePerGas),
+      maxPriorityFeePerGas: formatGwei(cancelGasPrice.maxPriorityFeePerGas),
+    });
+
+    // Send 0-value tx to self with higher gas price
+    const cancelTxHash = await this.walletClient.sendTransaction({
+      to: account.address,
+      value: 0n,
+      nonce,
+      gas: 21_000n, // Standard ETH transfer gas
+      maxFeePerGas: cancelGasPrice.maxFeePerGas,
+      maxPriorityFeePerGas: cancelGasPrice.maxPriorityFeePerGas,
+    });
```

Comment:
Wondering if we should just send, or send and monitor?

Reply:
in an unfortunate scenario, could this end up creating another timed-out tx that then requires cancellation and so on? 😅

Reply:
I mean, it'd be the "same" tx, since it uses the same nonce, and it only gets created on timeout. But we could add a check that if the tx is a cancellation, then don't try cancelling.